packages/blade/src/components/Table/Table.web.tsx

```diff
@@ -0,0 +1,539 @@
+   * The function is called with an object that has a sortKey property that is the key of the column that is sorted and a isSortReversed property that is a boolean that determines whether the sort is reversed or not.
+   **/
+  onSortChange?: ({
+    sortKey,
+    isSortReversed,
+  }: {
+    sortKey: TableHeaderCellProps['headerKey'];
+    isSortReversed: boolean;
+  }) => void;
+  /**
+   * The sortFunctions prop is an object that has a key for each column that is sortable.
+   * The value of each key is a function that is called when the column is sorted.
+   * The function is called with an array of the rows in the table.
+   * The function should return an array of the rows in the table.
+   **/
+  sortFunctions?: Record<string, (array: TableNode<Item>[]) => TableNode<Item>[]>;
+  /**
+   * The toolbar prop is a React element that is rendered above the table.
+   * The toolbar prop should be a `TableToolbar` component.
+   **/
+  toolbar?: React.ReactElement;
+  /**
+   * The pagination prop is a React element that is rendered below the table.
+   * The pagination prop should be a `TablePagination` component.
+   **/
+  pagination?: React.ReactElement;
+  /**
+   * The height prop is a responsive styled prop that determines the height of the table.
+   **/
+  height?: BoxProps['height'];
+  /**
+   * The showStripes prop determines whether the table should have striped rows or not.
+   * The default value is `false`.
+   **/
+  showStripes?: boolean;
+  /**
+   * The gridTemplateColumns prop determines the grid-template-columns CSS property of the table.
+   * The default value is `repeat(${columnCount},minmax(100px, 1fr))`.
+   **/
+  gridTemplateColumns?: string;
+  /**
+   * The surfaceLevel prop determines the surface level of the table.
+   * The surfaceLevel prop can be 1, 2, 3, 4 or 5.
+   * The default value is `2`.
+   **/
+  surfaceLevel?: SurfaceLevels;
+  /**
+   * The isLoading prop determines whether the table is loading or not.
+   * The default value is `false`.
+   **/
+  isLoading?: boolean;
+  /**
+   * The isRefreshing prop determines whether the table is refreshing or not.
+   * The default value is `false`.
+   **/
+  isRefreshing?: boolean;
+} & StyledPropsBlade;
+
+const rowSelectType: Record<NonNullable<TableProps<unknown>['selectionType']>, SelectTypes> = {
+  single: SelectTypes.SingleSelect,
+  multiple: SelectTypes.MultiSelect,
+};
+
+// Get the number of TableHeaderCell components.
+// This is very complicated but the only way to iterate through the structure and get number of header cells.
+// Assuming number of header cells is the same as number of columns
+const getTableHeaderCellCount = (children: (data: []) => React.ReactElement): number => {
+  const tableRootComponent = children([]);
+  if (tableRootComponent && React.isValidElement(tableRootComponent)) {
+    const tableComponentArray = React.Children.toArray(tableRootComponent);
+    if (React.isValidElement(tableComponentArray[0])) {
```

Comment:
nit: see if we can use guard clauses / early returns to reduce the indentation here. 
Nested ifs right now making the code hard to read.